[Monitor Query] Add samples/docs for running big queries and overcoming API limits for size and record count (#35704)
Azure/azure-sdk-for-java: sdk/monitor/azure-monitor-query/README.md

}
```

#### Overcome Log Analytics query size limitations

> If your query exceeds the [service limits][service_limits], see the large log query documentation to learn how to overcome them.

Author: I'll add this after this PR merges.

}
```

#### Overcoming Log Analytics query size limitations

> If your query exceeds the service limits, you can refer to the large log query documentation to learn how to overcome them.

Reviewer: it's likely hard to add a link right away without failing the pipeline (only possible to add a permalink to your fork), but as a follow-up PR, mind adding a link to the other doc?

> #### Overcoming Log Analytics query size limitations
> If your query exceeds the service limits, you can refer to the large log query documentation to learn how to overcome them.

Reviewer 2: As a rule of thumb, avoid gerunds in headings. Also, this "large log query documentation" text should be a link.
```suggestion
#### Overcome Log Analytics query size limitations

If your query exceeds the service limits, see the large log query documentation to learn how to overcome them.
```

Author: @scottaddie I plan on linking via a GitHub permalink once the PR merges.

> If your query exceeds the service limits, you can refer to the large log query documentation to learn how to overcome them.

Author: @srnagar I can't link to the documentation I added until the PR gets merged because it'll fail the ci. I'm going to have a follow-up linking to the documentation that is in the samples README file. I'll update this text to reflect that.
